cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaSetOperator.java

```diff
@@ -49,4 +58,64 @@ public static boolean needsRollingUpdate(StatefulSetDiff diff) {
         }
         return false;
     }
+
+    private void revertVolumeChanges(StatefulSet current, StatefulSet desired) {
+
+        Container currentKafka =
+                current.getSpec().getTemplate().getSpec().getContainers().stream().filter(c -> c.getName().equals("kafka")).findFirst().get();
+        Container desiredKafka =
+                desired.getSpec().getTemplate().getSpec().getContainers().stream().filter(c -> c.getName().equals("kafka")).findFirst().get();
+
+        desiredKafka.setVolumeMounts(currentKafka.getVolumeMounts());
+
+        StatefulSetBuilder ssBuilder = new StatefulSetBuilder(desired);
+        StatefulSet updated = ssBuilder
+                .editSpec()
+                    .editTemplate()
+                        .editSpec()
+                            .editFirstContainer()
+                                .editMatchingEnv(e -> e.getName().equals(KafkaCluster.ENV_VAR_KAFKA_LOG_DIRS))
+                                    .withValue(desiredKafka.getVolumeMounts().stream()
+                                    .filter(vm -> vm.getMountPath().contains(AbstractModel.VOLUME_NAME))
+                                    .map(vm -> vm.getMountPath()).collect(Collectors.joining(",")))
+                                .endEnv()
+                            .endContainer()
+                        .endSpec()
+                    .endTemplate()
+                .endSpec()
+                .build();
+
+        desired.setSpec(updated.getSpec());
+    }
+
+    @Override
+    protected StatefulSetDiff revertStorageChanges(StatefulSet current, StatefulSet desired) {
+
+        List<PersistentVolumeClaim> currentPvcs = current.getSpec().getVolumeClaimTemplates();
+        List<PersistentVolumeClaim> desiredPvcs = desired.getSpec().getVolumeClaimTemplates();
+
+        if (desiredPvcs.size() != currentPvcs.size()) {
+            log.warn("Adding or removing Kafka persistent storage is not possible. The changes will be ignored.");
+            revertVolumeChanges(current, desired);
+        } else {
+
+            for (PersistentVolumeClaim currentPvc : currentPvcs) {
+
+                Optional<PersistentVolumeClaim> pvc =
+                        desiredPvcs.stream().filter(desiredPvc -> desiredPvc.getMetadata().getName().equals(currentPvc.getMetadata().getName())).findFirst();
```

Comment:
Again the `.findFirst()` on a new line helps when reading the streaming operations. Also this is a long line!